src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -698,9 +699,47 @@ export default class QuestionnaireForm extends Component {
     return c;
   }
 
+  addAuthorToResponse(qr, practitionerRef) {
+    function traverseToItemsLeafNode(item, practitionerRef) {
+      if (!item.item) {
+        return addAuthor(item, practitionerRef);
+      }
+      else {
+        item.item.map(item => {
+          traverseToItemsLeafNode(item, practitionerRef);
+        })
+      }
+    }
+    // url is a string
+    function addAuthor(item, practitionerRef) {
+      var url = "http://hl7.org/fhir/StructureDefinition/questionnaireresponse-author"
+      const urlValRef =
+      {
+        "url": url,
+        "valueReference":
+        {
+          "reference": practitionerRef
+        }
+      }
+      if (item.extension) {
+        const found = item.extension.find(element => element.url === url)
+        // if there is already an extension with author-extension url
+        if (!found) {
```

Comment:
The issue is that even if we find the same url, the valueReference could be different from urlValRef, we need to overwrite it with the new value. The solution is to find the index of the existing item if found, then put the new value to that location.